### PR TITLE
Refactor tests and code in retrying

### PR DESCRIPTION
**Pull request by ArtDu**
Coverage:
image

I haven't forgotten about:
- [x] Tests
- [ ] Changelog
- [ ] Documentation
- [x] Commit messages comply with the [guideline](https://www.tarantool.io/en/doc/latest/dev_guide/developer_guidelines/#how-to-write-a-commit-message)
- [x] Cleanup the code for review. See [checklist](https://github.com/tarantool/cartridge-java/blob/master/docs/review-checklist.md)

Related issues:

### CODE REVIEW

File: src/main/java/io/tarantool/driver/api/retry/TarantoolRequestRetryPolicies.java [hunk 1]
```diff
@@ -138,15 +138,9 @@ public <R> CompletableFuture<R> wrapOperation(Supplier<CompletableFuture<R>> ope
...
-                                if (lastException != null) {
-                                    resultFuture
-                                        .completeExceptionally(
-                                            new TarantoolTimeoutException(operationTimeout, lastException));
-                                } else {
-                                    resultFuture
-                                        .completeExceptionally(new TarantoolTimeoutException(operationTimeout));
-                                }
+                                resultFuture
+                                    .completeExceptionally(
```
Comment (Elishtar) [inline src/main/java/io/tarantool/driver/api/retry/TarantoolRequestRetryPolicies.java:142]: May by better resultFuture.completeExceptionally(

File: src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java [hunk 2]
```diff
@@ -75,6 +98,66 @@ void testUnboundRetryPolicy_returnSuccessAfterFailsWithDelay() throws ExecutionE
...
+        assertEquals(expectedCallback, policyFactory.getCallback());
+    }
+
+    @Test
+    void testUnboundRetryPolicy_shouldHandleCorrectly_ifOperationThrowException() throws InterruptedException {
+        RequestRetryPolicy policy = TarantoolRequestRetryPolicies.unbound().withDelay(10).build().create();
+        CompletableFuture<Boolean> wrappedFuture = policy.wrapOperation(this::operationWithException, executor);
+        try {
+            wrappedFuture.get();
+        } catch (ExecutionException e) {
```
Comment (Elishtar) [inline src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java:115]: May be `assertThrows`?

File: src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java [hunk 3]
```diff
@@ -75,6 +98,66 @@ void testUnboundRetryPolicy_returnSuccessAfterFailsWithDelay() throws ExecutionE
...
+    @Test
+    void testUnboundRetryPolicyCanRetryRequest_shouldWorkCorrectly() throws InterruptedException {
+        AtomicInteger counter = new AtomicInteger(2);
+        RequestRetryPolicy policy = TarantoolRequestRetryPolicies.unbound(
+            ex -> counter.getAndDecrement() > 0
+        ).build().create();
+        CompletableFuture<Boolean> wrappedFuture = policy.wrapOperation(this::simpleFailingFuture, executor);
+        try {
+            wrappedFuture.get();
+        } catch (ExecutionException e) {
```
Comment (Elishtar) [inline src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java:139]: May be assertThrows?

File: src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java [hunk 4]
```diff
@@ -240,6 +336,14 @@ private CompletableFuture<Boolean> simpleFailingFuture() {
         return result;
     }
 
+    private CompletableFuture<Boolean> neverCompletedFuture() {
```
Comment (Elishtar) [inline src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java:339]: Let's try to use @Mock

**Review by akudiyar — CHANGES_REQUESTED** [review]
Overall LGTM, but I agree with the comments from Andrey.

File: src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java (hunk 4 above)
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java:339]: I don't see an easy way to use Mock here. Cause we need to mock get method for sleeping and it looks more complicated

File: src/main/java/io/tarantool/driver/api/retry/TarantoolRequestRetryPolicies.java (hunk 1 above)
Comment (ArtDu) [reply src/main/java/io/tarantool/driver/api/retry/TarantoolRequestRetryPolicies.java]: Fixed

File: src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java (hunk 2 above)
Comment (ArtDu) [reply src/test/java/io/tarantool/driver/api/retry/RequestRetryPolicyTest.java]: Replaced all try catch assert on it

**Review by akudiyar — APPROVED** [review]
LGTM

**Pull request merged**